Classes/DDOSLogger.m

```diff
@@ -17,10 +17,49 @@
 
 #import <os/log.h>
 
+@interface DDOSLogger () {
+    NSString *_subsystem;
+    NSString *_category;
+}
+@property (copy, nonatomic, readonly) NSString *subsystem;
+@property (copy, nonatomic, readonly) NSString *category;
+@property (strong, nonatomic, readwrite) os_log_t  logger;
+@end
+
 @implementation DDOSLogger
 
+@synthesize subsystem = _subsystem;
+@synthesize category = _category;
+
+#pragma mark - Initialization
+
+/**
+ * Assertion
+ * Swift: (String, String)?
+ */
+- (instancetype)initWithSubsystem:(NSString *)subsystem category:(NSString *)category {
+    NSAssert((subsystem == nil) == (category == nil), @"Either both subsystem and category or neither can be nil.");
+    if (self = [super init]) {
+        _subsystem = [subsystem copy];
+        _category = [category copy];
+    }
+    return self;
+}
+
 static DDOSLogger *sharedInstance;
 
+- (instancetype)init {
+    if (sharedInstance != nil) {
```

Comment:
Why should it be wrong to create another DDOSLogger if we have a shared instance? The other loggers also allow multiple instances besides the shared instance.

Reply:
@ffried 
I suppose that you suggest to completely remove `sharedInstance`?
But even `DDTTYLogger` has the same logic in `-init`.

Reply:
`sharedInstance` is fine by me for those who want to use it. But we shouldn't ban those that don't want to use it.
I'm usually comparing with DDFileLogger, since this is currently the only logger that isn't deprecated 😉

Reply:
@ffried 
`DDOSLogger` already has `sharedInstance` of `OS_LOG_DEFAULT` inside. We have correct lookup in `DDOSLogger` for system `os_log` no matter how to store `DDOSLogger`. Maybe just remove this `sharedInstance`?

Reply:
We could. But this would be a breaking change.

Reply:
For now this is fine, I think.